internal/tiltfile/telemetry/telemetry.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+
+	"github.com/windmilleng/tilt/internal/tiltfile/starkit"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+type Settings struct {
+	Cmd model.Cmd
+}
+
+type Extension struct {
+}
+
+func NewExtension() Extension {
+	return Extension{}
+}
+
+func (e Extension) NewState() interface{} {
+	return Settings{
+		Cmd: model.Cmd{},
+	}
+}
+
+func (Extension) OnStart(env *starkit.Environment) error {
+	return env.AddBuiltin("experimental_telemetry_cmd", setTelemetryCmd)
+}
+
+func setTelemetryCmd(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var cmd string
+	err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs, "cmd", &cmd)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	if len(cmd) == 0 {
+		return starlark.None, fmt.Errorf("cmd cannot be empty")
+	}
+
+	var innerErr error
+	err = starkit.SetState(thread, func(settings Settings) Settings {
```

Comment:
The SetState callback is allowed to return an error, see how config.Extension does it